### PR TITLE
Update json-schema >=3.0.0 <=6.0.0, and atoum >= 3.0

### PR DESCRIPTION
**Pull request by shouze**

### CODE REVIEW

**Comment by shouze** [comment]
ping @jubianchi @Hywan 

Also, looks like all dataProviders randomly leads to some kind of infinite loop (especially `testValidatesJsonArrayGrammar` & `testNotValidatesJsonArrayGrammarDataProvider`).

As you can see half of ci matrix timeout because of that.

I've investigated furthermore and it's related to `$this->sampleMany()` and maybe also to `*.pp` files. When I limit sampleMany to 1 it occurs less often but still occurs.
So the only thing I'm sure is that it's related to `atoum/praspel-extension` whatever:
- php version.
- composer update `--prefer-lowest` or `--prefer-stable`.

File: composer.json [hunk 1]
```diff
@@ -13,15 +13,15 @@
...
+        "atoum/atoum": "^3.0",
+        "justinrainbow/json-schema": ">=3.0.0 <=6.0.0"
     },
     "require-dev": {
-        "atoum/praspel-extension": "^0.16",
+        "atoum/praspel-extension": "^0.16||^0.17",
         "hoa/praspel": "^1.16.01.14",
         "hoa/compiler": "^3.16.08.15",
-        "hoa/Iterator": "2.16.03.15"
+        "hoa/Iterator": "^2.16.03.15"
```
Comment (Hywan) [inline composer.json]:
```diff
- hoa/Iterator
+ hoa/iterator
```

**Comment by Hywan** [comment]
What computer and OS do you have please?

**Comment by shouze** [comment]
@Hywan macOS 10.13 with php 7.1.7 but the same thing occurs on ci with various php releases on Ubuntu Trusty.

**Comment by shouze** [comment]
I've also tested with & without opcache & xdebug exts, don't change anything.

**Comment by shouze** [comment]
@Hywan did you suspect an issue about case sensitivity? Because after fixing the hoa/iterator requirement looks ok on ci.

Also... I didn't removed hhvm support but anyway travis says it's time to switch to trusty if we want to test hhvm.

**Comment by shouze** [comment]
ping @Hywan so are you willing to merge or any other required changes?

FYI I'm still experiencing the issue on my local MacOS php. php process take 100% CPU, it's an infinite loop for sure.

If I quickly run `sudo dtruss -p PID` I get that:
```dtrace
dtrace: error on enabled probe ID 2721 (ID 176: syscall::wait4:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2389 (ID 840: syscall::fstat64:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2166 (ID 168: syscall::read:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2389 (ID 840: syscall::fstat64:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2166 (ID 168: syscall::read:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2721 (ID 176: syscall::wait4:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2389 (ID 840: syscall::fstat64:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2166 (ID 168: syscall::read:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2389 (ID 840: syscall::fstat64:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2166 (ID 168: syscall::read:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2721 (ID 176: syscall::wait4:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2389 (ID 840: syscall::fstat64:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2166 (ID 168: syscall::read:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2389 (ID 840: syscall::fstat64:return): invalid user access in action #5 at DIF offset 0
dtrace: error on enabled probe ID 2166 (ID 168: syscall::read:return): invalid user access in action #5 at DIF offset 0
```

**Comment by Hywan** [comment]
I can reproduce the infinite loop, but only when testing, it does not happen with the extension. So let's merge, and investigate later.

**Comment by Hywan** [comment]
I've release the 2.0.0 version, https://github.com/atoum/json-schema-extension/releases/tag/2.0.0.

**Comment by shouze** [comment]
@Hywan yes that's it, testing related infinite loop only.

**Pull request merged**